File: implementations/dart/test/id57_short_test.dart
import 'package:test/test.dart';
import 'package:b57/b57.dart';

void main() {
  group('ID57-SHORT', () {
    test('default and verify', () {
      final s = id57ShortGenerate('abc'.codeUnits, ID57ShortLength.def);
      expect(id57ShortVerify('abc'.codeUnits, s, ID57ShortLength.def), true);
    });

    test('isValid and isCanonical', () {
      final s = id57ShortGenerate('test'.codeUnits, ID57ShortLength.def);
                });
  });
}

File: implementations/go/id57_short_test.go
package b57

import "testing"

func TestID57ShortGenerate(t *testing.T) {
	input := []byte("hello id57 short")
	h, err := ID57ShortGenerate(input, ID57ShortDefault)
	if err != nil {
		t.Fatalf("expected nil err, got %v", err)
	}
	if !ID57ShortVerify(input, h, ID57ShortDefault) {
		t.Fatalf("verify failed")
	}
}
